Backend/Models/Users/UserManager.cs

```diff
@@ -72,9 +72,22 @@ public static bool Authenticate(WebSocket socket, string username)
                 return false;
             }
 
-            user.Username = username;
+            User? userInDb = User.GetUserFromDB(username);
+            if (userInDb == null)
+            {
+                user.Username = username;
+                user.SaveToDb();
+            }
+            else
+            {
+                user = userInDb;
+            }
             user.IsRegistered = true;
-            user.SaveToDb();
+
+            if (user.IsBanned)
+            {
+                Disconnect(user);
```

Comment:
We should probably return false after this disconnect and log that the user is banned but I guess this is part of the banned work?

Reply:
nah you're right, ill get on with it